README
README for Rubric 0.148

Rubric is a note-keeping system that also serves as a bookmark manager.  Users
store entries, which are small (or large) notes with a set of categorizing
"tags."  Entries may also refer to URIs.

--- WARNING ---

This is young software, likely to have bugs and likely to change in strange
ways.  I will try to keep the documented API stable, but not if it makes
writing Rubric too inconvenient.

Basically, just take note that this software works, but it's still very much
under construction.

--- INSTALLING ---

Things you'll have to do to use it:

  1. install the modules with the usual "make install"
  2. put rubric.cgi and rubric.yml somewhere together in your docroot
     (start with the rubric.yml in etc/)
  3. create a location in which to put rubric.db
  4. update rubric.yml (if you require verification, set email_from)
  5. in the dir with rubric.yml, run "rubric db --setup" to generate rubric.db
  6. set rubric.db +rw by the httpd user; set its location +rwx
  7. create a user object for yourself (via the web or with adduser.pl)
  8. (optional) set up .htaccess to prevent access to db, yml, and templates
  9. go?

--- UPGRADING ---

Things you'll have to do:

  1. install the new templates (and probably the css)
  2. install the modules
  3. run rubric db --update (from the location of rubric.cgi/.yml)
  4. update your config

--- OTHER INFORMATION ---

To import your bookmarks from del.icio.us:

  1. get my delicious-backup script (on the CPAN)
  2. dump your bookmarks to YAML (delbackup -y)
  3. run:  load_yml USERNAME < bookmarks.yml




Changes
Revision history for Rubric

{{$NEXT}}
        - update metadata: dist no longer supported by author

0.156     2015-12-16 21:48:57-05:00 America/New_York
        - add cookie_secure, cookie_httponly, and secure_login
          to make it easier to keep your Rubric login data out of the clear

0.155     2014-03-20 18:48:28-04:00 America/New_York
        - improve chances of producing valid XHTML
        - allow the instance to be renamed with "basename"

0.154     2013-10-15 22:30:06 America/New_York
          update bugtracker and repo links to point at GitHub

0.153     2012-12-24 23:17:19 America/New_York
          use Email::Sender, not Email::Send (breaks smtp_server config
          for now)

0.152     2012-07-10 16:29:38 America/New_York
          add a missing prereq, stop losing errors doing session decrypt
          (thanks for the report, Jan Dubois!)

0.151     2012-02-24 10:18:29 America/New_York

          switched from YAML to YAML::XS

          now built with Dist::Zilla; some docs may be a bit weird until I go
          through them with a fine-toothed comb

0.150     2012-02-24 10:09:47 America/New_York

          TOTALLY BUSTED; DO NOT USE

0.149     2011-10-23

          more properly handle encoding for entry text/title/description --
          still a mess, but much, much more likely to just work, now


0.148     2011-08-22

          fix a bug that prevented `rubric user --new-user` from working

          include everything needed to run under PSGI -- although app.psgi
          is not currently installed anywhere

0.147     2011-02-02

          [MIGHT BREAK EVERYTHING]
          remove Config::Auto and stop using it; might break everything!

0.146     2011-01-08
          do not allow Encode to destructively process forms

0.145     2010-01-11
          do not require users to have JSON::XS

0.144     2009-03-11
          add entry title to <title> tag (bricas)
          use String::TagString

0.143     2007-07-01
          minor testing/packages updates
          add calendar view (jcap)
          add tag cloud (jcap)
          do work to make things work under fcgi (rjbs)

0.142     2006-09-05
          update install instructions

0.141     2006-09-02
          freshen Module::Install

0.140     2006-05-18
          new version numbering, new release schedule: less stable, more often
          fix a small RSS bug that bugged a large number Firefox users
          lots of forgotten changes

0.12      2005-10-26 (long overdue)
          some tiny test cleanups that might make tests... pass!
          moved repository to svn
          fixed stupid handling of repost to URIs with one-entry-per-uri config
          delete unneeded sessions at teardown (lousy?)
          work on unbreaking tags/facets
          rolled back Data::FormValidator use
          provide link on related tags to add/remove user constraint
          show related tags even when no user selected
          QUESTIONABLE DESCISION: @-prefix tags don't list in tagboxes
          list user's name for "your tags" if user ne current_user
          begin stubbing out pluggable body rendering
          begin stubbing out facets
          allow query for "/exact_tags/" to work (find untagged)
          re-allow "+" as tag separator in tagstrings

0.10      2005-06-01
          a little more and better testing
          fixed broken 'help' runmode

0.09_09   2005-05-29
          enable utf-8 tags
          fix missing-title error on blank new post form
          simplify updatedb script, eliminate vacuumdb script
          display_localtime option

0.09_08   2005-05-28
          private_tag implementation no longer completely disables public entries
          expand/collapse link now preserves page number
          help page's bookmarklet now uses encodeURIComponent, not escape
          added an input box for searching

0.09_07   2005-05-27
          private_tag implementation no longer horribly breaks entry listing
          fixes to work with DBD::SQLite 1.08, now required

0.09_06   2005-05-27
          added private_tag - SO SETTLE DOWN, SSORICHE!

0.09_05   2005-05-25
          fixed bug that snuck back into _04, reset_password was broken
          introduced Rubric::DBI::Setup for db maintenance

0.09_04   2005-05-24
          refactor get_* to return property, not WebApp
          add stubby password-reset functionality

0.09_03   2005-05-02
          fix tag-destroying bug

0.09_02   2005-04-29
          basic experimental searching
          hyphens allowed as non-first char in tag
          uri_root defaults to "" not undef

0.09_01   2005-04-12
          describe queries (badly)
          redirect to previous query after deleting a link
          validate that post input is utf-8
          add first_only query constraint
          fix RSS more: use rdf:about attr as intended!
          taglists tags are now "click to add" on post form
          added "tags for this link"
          if posting a non-unique uri, previous post defaults in (old bug?)

0.08      2005-04-07
          use normal must-login/redirect-on-login behavior for /preferences
          honor requirement for verification on unverified users
          use html_line_breaks in RSS feed for paragraphing
          POD corrections

0.07_07   2005-04-04
          added exact_tags query string
          +/- filter on taglist centers on gecko; gecko bug must die
          updated "related tags" taglist for clarity of add/remove tags
          RSS should now validate and work everywhere (I think)
          expand/collapse entry bodies on default entries templates
          the numeric tag "0" now works
          the (edit) links will now include a redirect to your previous query
          /edit or /entry links to missing entries (or non-owned for edit)
            produce better error messages
          max/default entries per page is now configurable

0.07_06   2005-04-02

Makefile.PL
use strict;
use warnings;

use inc::Module::Install;
use File::Find;

name          ('Rubric');
module_name   ('Rubric');
author        ('Ricardo SIGNES <rjbs@cpan.org>');
license       ('perl');
version_from  ('lib/Rubric.pm');

requires('App::Cmd'          => 0.000);
requires('CGI::Application'  => 3.00);
requires('CGI::Application::PSGI' => 0);
requires('CGI::Carp'         => 0.00);
requires('Class::DBI'        => 0.96);
requires('Class::DBI::utf8'  => 0.2); # all I tested with
requires('Crypt::CBC'        => 0); # min version unknown
requires('Crypt::Rijndael'   => 0); # min version unknown
requires('DateTime'          => 0.00);
requires('Date::Span'        => 1.12);
requires('DBD::SQLite'       => 1.08);
requires('Digest::MD5'       => 0.00);
requires('Email::Address'    => 0.00);
requires('Email::Send'       => 0.00);
requires('Encode'            => 2.00);
requires('JSON'              => 2); # min version unknown
requires('File::ShareDir'    => 0); # min version unknown
requires('MIME::Base64'      => 0); # min version unknown
requires('Plack::Builder'    => 0); # min version unknown
requires('String::TagString' => 0.00);
requires('String::Truncate'  => 0.00);
requires('Sub::Exporter'     => 0); # min version unknown
requires('Template'          => 2.00);
requires('Time::Local'       => 0.00);
requires('Time::Piece'       => 0.00);
requires('YAML'              => 0.00);
requires('Class::DBI::AbstractSearch' => 0);
requires('HTML::CalendarMonth'        => 0.00);
requires('HTML::TagCloud'             => 0.00);
requires('HTML::Widget::Factory'      => '0.03'); # 0.03: links w/classes
requires('HTTP::Server::Simple'       => '0.09');
requires('Getopt::Long::Descriptive'  => '0.03');
requires('Template::Plugin::Class'    => '0.12');
requires('Test::HTTP::Server::Simple' => '0.02');
requires('Test::WWW::Mechanize'       => '1.04');

install_script('bin/rubric');

install_share;

tests_recursive;

WriteAll();
